membership-attribute-service/app/controllers/AttributeController.scala

```diff
@@ -61,54 +69,78 @@ class AttributeController(attributesFromZuora: AttributesFromZuora, commonAction
     else Future.successful(None)
   }
 
+  def getLatestMobileSubscription(identityId: String)(implicit executionContext: ExecutionContext): Future[Option[MobileSubscriptionStatus]] = {
+    mobileSubscriptionService.getSubscriptionStatusForUser(identityId).transform {
+      case Failure(NonFatal(error)) =>
+        log.warn("Exception while fetching mobile subscription, assuming none", error)
+        Success(None)
```

Comment:
is it a problem is this stops working? if so how would we know? is it worth pushing a metric and alarming on it?

Reply:
>is it a problem is this stops working
Yes
>if so how would we know?
We wouldn't

Should I remove the handling of the exception and let it bubble up?

Reply:
bubbling would cause a 5xx error and get no useful data at all. So we would know about it, but it would be more severe in terms of breaking swimlaning. I like that you've made an error barrier, I would just like an alert as well.

I'm not surprised if the existing code isn't too smart on this, and I'm happy to have alerting in a subsequent PR (it's not blocking this one)

My suggestion would be to push a metric to cloudwatch and add an alarm on that.
Here is a ready made implementation example for infrequent metrics: https://github.com/guardian/support-frontend/blob/master/support-services/src/main/scala/com/gu/aws/AwsCloudWatchMetricPut.scala